### PR TITLE
Add targets that were missing in rustc

### PR DESCRIPTION
**Pull request by Sycration**
The 1.51 release notes said that i386 and i486 linux were supported, but it was not included in rustc. This adds those.

### CODE REVIEW

File: compiler/rustc_target/src/spec/mod.rs [hunk 1]
```diff
@@ -663,6 +663,8 @@ supported_targets! {
     ("x86_64-unknown-linux-musl", x86_64_unknown_linux_musl),
     ("i686-unknown-linux-musl", i686_unknown_linux_musl),
     ("i586-unknown-linux-musl", i586_unknown_linux_musl),
+    ("i386_unknown_linux_gnu", i386_unknown_linux_gnu),
+    ("i486_unknown_linux_gnu", i486_unknown_linux_gnu),
```
Comment (jonas-schievink) [inline compiler/rustc_target/src/spec/mod.rs:667]:
```suggestion
    ("i386-unknown-linux-gnu", i386_unknown_linux_gnu),
    ("i486-unknown-linux-gnu", i486_unknown_linux_gnu),
```

Comment (Sycration) [reply compiler/rustc_target/src/spec/mod.rs:667]: Sorry, Idk how I missed that

**Comment by jonas-schievink** [comment]
You'll have to add these to https://github.com/rust-lang/rust/blob/master/src/doc/rustc/src/platform-support.md as Tier 3 targets, so that CI will pass

**Comment by Mark-Simulacrum** [comment]
Hm, so before we merge this I would like to see the tier 3 policy [requirements](https://doc.rust-lang.org/nightly/rustc/target-tier-policy.html#tier-3-target-policy) explicitly documented (e.g., in a comment here and a list of the maintainers).

You can see a sample of how this might be done in https://github.com/rust-lang/compiler-team/issues/428#issuecomment-850307216, though note that MCP is seeking a slightly higher bar (Tier 2). That particular comment just addresses tier 3 status, though.

**Comment by Sycration** [comment]
> Hm, so before we merge this I would like to see the tier 3 policy [requirements](https://doc.rust-lang.org/nightly/rustc/target-tier-policy.html#tier-3-target-policy) explicitly documented (e.g., in a comment here and a list of the maintainers).
> 
> You can see a sample of how this might be done in [rust-lang/compiler-team#428 (comment)](https://github.com/rust-lang/compiler-team/issues/428#issuecomment-850307216), though note that MCP is seeking a slightly higher bar (Tier 2). That particular comment just addresses tier 3 status, though.

Would [this](https://gist.github.com/Sycration/cbcc82c3023af15d9fdbde6af48c3f91) be acceptable? I'm still a newbie at this.

**Comment by joshtriplett** [comment]
Yeah, these targets absolutely need separate maintainers listed. Just because someone maintains i686 targets does not mean it's safe to assume they want to maintain these much older, less widely used targets.

(For that matter, we should probably determine who the maintainers of the i586 targets are, which is likely separate from the maintainers of i686.)

**Comment by joshtriplett** [comment]
Moving this to "waiting-on-author", as it needs maintainers in order to move forward.

**Comment by JohnCSimon** [comment]
Added `S-blocked` because it's waiting on maintainers. 
cc @Sycration @joshtriplett

**Comment by JohnCSimon** [comment]
@Sycration  
Ping from triage: I'm closing this due to inactivity, Please reopen when you are ready to continue with this. 
Note: if you do please open the PR **BEFORE** you push to it, else you won't be able to reopen - this is a quirk of github. 
Thanks for your contribution.

@rustbot label: +S-inactive

**Pull request closed without merging**